core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1230,6 +1176,23 @@ public PercolateStats percolateStats() {
         return percolatorQueriesRegistry.stats();
     }
 
+    /**
+     * Asynchronously refreshes the engine for new search operations to reflect the latest
+     * changes.
+     */
+    public void refreshAsync(final String reason) {
+        engineConfig.getThreadPool().executor(ThreadPool.Names.REFRESH).execute(new Runnable() {
```

Comment:
we do this already in `EngineRefresher` in this class, maybe we can factor it out and put that here?